### DISCUSSION TITLE
Doric cube 2nd gen with LED and photodetecor

### DISCUSSION THREAD

**Discussion by MartinaCireraFerrer** [pyPhotometry/code#17] [Q&A]
Hello,
For my internship, I must decide on a fiber photometry system (low-cost as it is a pilot project in a university lab). I have already purchased the Doric 3 port mini-cube with integrated LED and photodetector (single fluorophore: GFP, 2nd gen). I first wanted to use the Analog Discovery 2 for data acquisition but I still needed to find a LED driver. Then I stumbled upon this product. I have read that it is compatible with the Doric hardware. However, I see that the max output current is 100 mA (Doric driver goes up to 2000 mA in normal mode). Is anyone using this device with the same Doric cube? Do you have comments on its performance, recommendations, or ideas? I am talking with Doric to confirm that using foreign devices is safe for the LED, but I would like some input from people who have tried it in real life! 

I was also wondering. If I want Sinusoidally-Modulated Illumination, could I somehow connect the pyPhotometry board to my Analog Discovery 2 (which has pattern/wavegen modules that allow sinusoidal generation) as an interface? I would still use the pyPhotometry board for data acquisition and as a LED driver, but it would be controlled from my AD2. I hope it makes sense!

Thank you!

Martina

**Comment by ThomasAkam** [comment]
Hi @MartinaCireraFerrer,

PyPhotometry is compatible with Doric Gen 2 minicubes.

Typically low LED currents are used for photometry, and we have found <100mA sufficient for our experiments (mostly dlight and gcamp6).

The system does implement pulsed illumination but not sinusoidal, though you you could potentially modify the code to do sinusoidal.

I will be travelling till next Wednesday so won't be able to answer further questions untill then.

**Comment by ThomasAkam** [comment]
We recently purchased a new system based on a Doric Gen 2 minicube and much to my suprise when I tested it the light ouput from the 465nm LED was much lower than what we were getting from our old systems based on Doric minicubes without integrated LEDs.  After discussing this with Doric, it turns out that the Gen 2 minicubes have an intensity adjustment ring between the LED and the minicube body which dramatically reduces light coupling from the LED to the sample.  Luckily it is straightforward to remove this, giving an ~6x increase in light intensity at the sample.  I have added information to the [Optical Components docs](https://pyphotometry.readthedocs.io/en/latest/resources/optical-components/) about this, including instructions on how to remove the intensity adjustment ring.
